mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -61,6 +62,9 @@
       }
     }
 
+    String[] readColumns = ColumnProjectionUtils.getReadColumnNames(job);
+    job.setStrings(InputFormatConfig.COLUMN_PROJECTIONS, readColumns);
```

Comment:
This is a bit different to how it was originally implemented in Hiveberg (https://github.com/ExpediaGroup/hiveberg/pull/14/files) which didn't require an additional property. Was this something that changed after that PR was merged?

Reply:
Yeah its slightly different after the changes in Iceberg with all the refactoring of the different IF's. So in Hiveberg we've only got the one IF and that's where we call planTasks() which is the part where you can project the columns. But in Iceberg now, the HiveIF is wrapped in the Mapred one and that extends the base IcebergIF which is where all the actual stuff to planTasks goes on. And that bit uses the fields from the InputFormatConfig, which is why I have to set it here^. I had to make a similar change for adding the filter pushdown

Reply:
OK, makes sense, go ahead and open this up on Apache and let's see what the others say.